### PR TITLE
Validate speedtest hotspots

### PR DESCRIPTION
**Pull request by maplant**

### CODE REVIEW

**Review by jeffgrunewald — APPROVED** [review]
only question again is whether or not we need the gateway to be _asserted_ in order to validate the reported speedtest and qualify for rewards or if the gateway just needs to registered to the chain

File: mobile_verifier/src/speedtests.rs [hunk 1]
```diff
@@ -89,13 +91,21 @@ impl SpeedtestRollingAverage {
             .await;
 
         futures::stream::iter(tests_by_publickey.into_iter())
-            .then(move |(pubkey, cell_speedtests)| async move {
-                let rolling_average = exec
-                    .fetch(&pubkey)
-                    .await?
-                    .unwrap_or_else(|| SpeedtestRollingAverage::new(pubkey.clone()));
-                Ok((rolling_average, cell_speedtests))
+            .then(move |(pubkey, cell_speedtests)| {
```
Comment (bbalser) [inline mobile_verifier/src/speedtests.rs:94]: why not just do a`filter` step before this then that eliminiates any that are not valid?

Comment (maplant) [reply mobile_verifier/src/speedtests.rs:94]: The client can return an Error

**Comment by maplant** [comment]
> 

I believe @madninja can verify, but it just needs to registered to the chain

**Review by bbalser — APPROVED** [review]

**Pull request merged**